### PR TITLE
Add Go Point2f type and update GetPerspectiveTransform()

### PR DESCRIPTION
**Pull request by eh-steve**
- Adds a Go `Point2f` struct with float32 X/Y fields
- Adds a C `Points2f` struct wrapper (with `Contour2f` alias)
- Updates `GetPerspectiveTransform()` to use new `[]Point2f` type instead of `[]image.Point` - this is an API-breaking change, but the original type signature did not match the C++ type signature, since the C++ expects a `cv::Point2f`.

Possibly addresses https://github.com/hybridgroup/gocv/issues/105

### CODE REVIEW

**Comment by deadprogram** [comment]
Hi @eh-steve thanks for the PR and sorry about the delay in getting you some feedback.

Whenever possible, I have tried to use the Go std lib for interfaces to OpenCV, then converting to the needed OpenCV type. This has made it generally easier when integrating with some other non-OpenCV but image related Go code.

This is why the current code converts to `std::vector<cv::Point2f>` in the current code.

Is there something specific that you need to do that this is preventing you from doing?

If so, then it might be better to add a separate function such as `GetPerspectiveTransform2f()` to avoid breaking the current API.

What do you think?

**Comment by eh-steve** [comment]
Yeah, I appreciate that Go’s `image.Point` type is handy and works well for most things which use integer pixels, but this function in particular often need floats rather than ints as inputs (e.g. when transforming between pixel space and a Cartesian or GPS coordinate system).

I’m happy to add a second function which uses floats to avoid breaking the API, but I think the new Point2f types will have to stay, since Go’s `image` package doesn’t have a float point type.

Do you want me to add a second function for float points?

**Comment by deadprogram** [comment]
> Do you want me to add a second function for float points?

Yes, please.

All good points (ducks) sounds like a plan.

File: imgproc.go [hunk 1]
```diff
@@ -1602,14 +1602,25 @@ func ApplyCustomColorMap(src Mat, dst *Mat, customColormap Mat) {
...
-// corresponding 4 point pairs.
+// corresponding 4 point pairs as image.Point.
 //
 // For further details, please see:
 // https://docs.opencv.org/master/da/d54/group__imgproc__transform.html#ga8c1ae0e3589a9d77fffc962c49b22043
 func GetPerspectiveTransform(src, dst []image.Point) Mat {
 	srcPoints := toCPoints(src)
 	dstPoints := toCPoints(dst)
-	return newMat(C.GetPerspectiveTransform(srcPoints, dstPoints))
+	return Mat{p: C.GetPerspectiveTransform(srcPoints, dstPoints)}
```
Comment (deadprogram) [inline imgproc.go]: Any reason to not use `newMat()` here to be able to use profiling?

File: imgproc_test.go [hunk 2]
```diff
@@ -1385,6 +1385,31 @@ func TestGetPerspectiveTransform(t *testing.T) {
 	}
 }
 
+func TestGetPerspectiveTransform2f(t *testing.T) {
+	src := []Point2f{
+		{0, 0},
+		{10, 5},
```
Comment (deadprogram) [inline imgproc_test.go]: Should these be various floating point values here so as to smoke test with more "realistic" data?

File: imgproc.go (hunk 1 above)
Comment (eh-steve) [reply imgproc.go]: Good catch, I missed that

File: imgproc_test.go (hunk 2 above)
Comment (eh-steve) [reply imgproc_test.go]: I've updated the test, although they definitely are interpreted as `float32`s in the struct literal, even if they don't have a decimal point

**Comment by deadprogram** [comment]
Thanks for all of the work on this PR @eh-steve now merging.

**Pull request merged**